src/asynqp/connection.py

```diff
@@ -60,9 +60,10 @@ def close(self):
 
         This method is a :ref:`coroutine <coroutine>`.
         """
-        self.sender.send_Close(0, 'Connection closed by application', 0, 0)
-        yield from self.synchroniser.await(spec.ConnectionCloseOK)
-        self.closed.set_result(True)
+        if not self.closed.done():
+            self.sender.send_Close(0, 'Connection closed by application', 0, 0)
+            yield from self.synchroniser.await(spec.ConnectionCloseOK)
+            self.closed.set_result(True)
```

Comment:
``connection.close()`` is special because it signals the end for all operations and mostly you will not want to process errors here. Do you honestly feel, that this block of code is good API:

```python
try:
    yield from self.connection.close()
except ConnectionLostError:
    pass
```
It it raises an error, how do you expect the user to react to it apart from ignoring? User closes the connection, I can't think of an case where he would want to reconnect while closing it.